core/src/main/java/org/apache/brooklyn/core/config/BasicConfigKey.java

```diff
@@ -184,7 +192,7 @@ public String getDescription() {
     protected boolean reconfigurable;
     protected ConfigInheritance typeInheritance;
     protected ConfigInheritance runtimeInheritance;
-    protected Predicate<? super T> constraint;
+    protected List<Predicate<? super T>> constraints;
```

Comment:
But this is the whole point of this change, expose the constraints as a list of string on the REST API. I thought about doing the change you are suggesting @neykov (looking into the `Predicates.and()`) but there is no way to get back the underlying predicates.
Reverting would mean to support only one predicate at a time which is a contradiction with the current YAML implementation (can specify more than one)